lib/puppet/provider/service/launchd.rb

```diff
@@ -181,10 +194,9 @@ def self.job_list
   # it is 10.6 or greater. This allows us to implement different plist
   # behavior for versions >= 10.6
   def has_macosx_plist_overrides?
-    @product_version ||= self.class.get_macosx_version_major
     # (#11593) Remove support for OS X 10.4 & earlier
     # leaving this as is because 10.5 still didn't have plist support
-    return true unless /^10\.[0-5]/.match(@product_version)
```

Comment:
I love how this accidentally disabled using plists for OS X 10.10 and 10.11, since they match the regex.